mne/preprocessing/artifact_detection.py

```diff
@@ -333,3 +334,189 @@ def _annotations_from_mask(times, art_mask, art_name):
+
+    annotated_intervals = [
+        [onset, onset + duration] for onset, duration in
+        zip(annotations.onset[keep_mask], annotations.duration[keep_mask])
+    ]
+
+    # Merge overlapping annotation intervals
+    # Pre-load `merged_intervals` with the first interval to simplify
+    # processing
+    merged_intervals = [annotated_intervals[0]]
+    for interval in annotated_intervals:
+        merged_interval_stop = merged_intervals[-1][1]
+        interval_start, interval_stop = interval
+
+        if interval_stop < merged_interval_stop:
+            # Current interval ends sooner than the merged one; skip it
+            continue
+        elif (interval_start <= merged_interval_stop and
+                interval_stop >= merged_interval_stop):
+            # Expand duration of the merged interval
+            merged_intervals[-1][1] = interval_stop
+        else:
+            # No overlap between the current interval and the existing merged
+            # time period; proceed to the next interval
+            merged_intervals.append(interval)
+
+    merged_intervals = np.array(merged_intervals)
+
+    # Now extract the actual break periods
+    break_onsets = []
+    break_durations = []
+
+    # Handle the time period up until the first annotation
+    if (raw.first_time < merged_intervals[0][0] and
+            merged_intervals[0][0] - raw.first_time >= min_duration):
+        onset = raw.first_time  # don't add t_start_after_previous here
+        offset = merged_intervals[0][0] - t_stop_before_next
+        duration = offset - onset
+        break_onsets.append(onset)
+        break_durations.append(duration)
+
+    # Handle the time period between first and last annotation
+    for idx, _ in enumerate(merged_intervals[1:, :], start=1):
+        this_start = merged_intervals[idx, 0]
+        previous_stop = merged_intervals[idx - 1, 1]
+        if this_start - previous_stop < min_duration:
+            continue
+
+        onset = previous_stop + t_start_after_previous
+        offset = this_start - t_stop_before_next
+        duration = offset - onset
+        break_onsets.append(onset)
+        break_durations.append(duration)
+
+    # Handle the time period after the last annotation
+    if (raw._last_time > merged_intervals[-1][1] and
+            raw._last_time - merged_intervals[-1][1] >= min_duration):
+        onset = merged_intervals[-1][1] + t_start_after_previous
+        offset = raw._last_time  # don't subtract t_stop_before_next here
+        duration = offset - onset
+        break_onsets.append(onset)
+        break_durations.append(duration)
+
+    # Finally, create the break annotations
+    break_annotations = Annotations(
+        onset=break_onsets,
+        duration=break_durations,
+        description=['BAD_break'],
+        orig_time=raw.info['meas_date']
+    )
+
+    logger.info(f'Detected {len(break_annotations)} break periods of >= '
+                f'{min_duration} sec duration.')
```

Comment:
I would add the total time annotated as well as total with something like
```suggestion
                f'{min_duration} sec '
                f *{sum(break_durations}:0.1f}/{raw.times[-1]:0.1f} sec total)')
```

Reply:
Also, the SI symbol for second is "s" and not "sec".

Reply:
I have modified the logging, it now generates output like:

```
Ignoring annotations with descriptions starting with: bad, edge
Detected 3 break periods of >= 20 sec duration:
    43.0 – 71.0 sec [28.0 sec]
    138.0 – 221.0 sec [83.0 sec]
    288.0 – 320.7 sec [32.7 sec]
In total, 51.7% of the data (143.7 sec) have been marked as a break.
```

@cbrnr We're using `sec` for logging in many other places of MNE as well, so I decided to use it here too for the sake of consistency. If you think we should change this, I'd appreciate if we could perhaps do it consistently across the entire code base in a separate PR?

Reply:
@larsoner I'm a bit at a loss whether I should take `raw.first_samp` into consideration when generating the logging output or not. I find it very confusing that I have to consider it when creating annotations, but `raw.plot()` doesn't set `show_first_samp=True` by default – now the logging output, based on `raw.annotations`, doesn't match what users see in the raw browser figure by default. elhp !

Reply:
I'm now subtracting `raw.first_time` from the annotation onsets I display in the log messages.